### ISSUE TITLE
Make read-write conflicts actually undefined in verilog emission

### ISSUE THREAD

**Issue by azidar** [freechipsproject/firrtl#496]
Right now, we generate a vector of registers, so the behavior is, while still legal, can be misleading for people.

@shunshou any additional comments?

**Comment by shunshou** [comment]
That's it. Just a scenario where this became a problem: 

I have a design that allows reads/writes to happen simultaneously (sometimes to the same address). I wrote some logic that theoretically passed the written value to data out when the address conflict occurred, but I actually screwed that up. It's just that w/ Xilinx + Chisel sims, I never caught the problem, but then the behavior with real SRAMs was undefined and tests that were passing ended up failing. Took some time to figure out inconsistencies. Would definitely prefer to have caught it earlier.
